src/thread/linked_list.cr

```diff
@@ -0,0 +1,64 @@
+require "thread/mutex"
+
+# :nodoc:
+class Thread
+  # :nodoc:
+  #
+  # Thread-safe doubly linked list of `T` objects that must implement
+  # `#previous : T?` and `#next : T?` methods.
+  struct LinkedList(T)
+    @mutex = Thread::Mutex.new
+    @head : T?
+    @tail : T?
+
+    # Iterates the list without acquiring the lock, to avoid a deadlock in
+    # stop-the-world situations, where a paused thread could have acquired the
+    # lock to push/delete a node, while still being "safe" to iterate (but just
+    # in this situation).
```

Comment:
"this situation" means that the world needs to be stoped, right?

Reply:
Yes.